modules/samplehold.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#ifndef DSY_SAMPLEHOLD_H
+#define DSY_SAMPLEHOLD_H
+
+#include <stdint.h>
+#ifdef __cplusplus
+
+namespace daisysp
+{
+/** Dual track and hold / Sample and hold module. \n 
+    Ported from soundpipe by Ben Sergentanis, June 2020.
+    @author Paul Batchelor
+    @date   2015
+*/
+class SampleHold
+{
+  public:
+    SampleHold() {}
+    ~SampleHold() {}
+
+    /** Track and hold circuit. Both circuits are updated in parallel.
+    \param gate Tracks new values as long as gate is true.
+    \param input Value to track.
+    \return Tracked and held value.
+    */
+    inline float Track(bool gate, float input)
+    {
+        Update(gate, input);
+        return sample_;
+    }
```

Comment:
I think both of `Track` and `Sample` could be changed to be just getters

i.e.
```C++
inline float Sample() const { return sample_; }
```

And then add a `Process` function that is basically your current `Update` function except public.

That function could take an argument for a track or sample (with a default) to make it a bit more idiomatic with the rest of the library.

```C++
// . . . In class
enum Mode
{
    MODE_SAMPLE_HOLD,
    MODE_TRACK_HOLD,
    MODE_LAST,
};

inline float Process(bool trigger, float input, Mode mode=MODE_SAMPLE_HOLD)
{
    Update(trigger, input);
    return mode == MODE_SAMPLE_HOLD ? sample_ : track_;
}
```

This will let you get the Sample/Track from the module without having to run the update function as well.